### PR TITLE
Release candidate for v1.4.0

### PR DESCRIPTION
**Pull request by tvogels01**
PRs on deck: 
* Upgrade to Python 3.5
* Add data lake to cloudformation
* Move configuration out of pipeline

### CODE REVIEW

File: bin/bootstrap.sh [hunk 1]
```diff
@@ -43,14 +43,17 @@ log () {
 log "Starting \"$0 $BUCKET_NAME $ENVIRONMENT\""
 set -x
 
-# Set creation mask to: u=rwx,g=rx,o=
-umask 0027
-
 # Install dependencies for Psycopg2, Arthur, and AWS shell commands we may run via datapipeline
 if [[ "$RUNNING_LOCAL" = "no" ]]; then
-    sudo yum install -y postgresql94-devel python34 python34-pip python34-virtualenv aws-cli gcc libyaml-devel tmux
+    sudo yum install -y postgresql95-devel python35 python35-pip python35-devel aws-cli gcc libyaml-devel tmux
```
Comment (bhtucker) [inline bin/bootstrap.sh]: I think now that `mypy` isn't installed here we no longer need a development build of Python

File: bin/bootstrap.sh [hunk 2]
```diff
@@ -76,19 +79,11 @@ if [[ "$RUNNING_LOCAL" = "no" ]]; then
...
-for VIRTUALENV in "virtualenv-3.4" "virtualenv"; do
-    if hash $VIRTUALENV 2> /dev/null; then
-        break
-    fi
-done
-
-# Make sure these steps match the description in the README!
-$VIRTUALENV --python=python3 venv
-# venv/bin/pip install pip --upgrade --disable-pip-version-check
+# Create virtual env for ETL -- Make sure these steps match the description in the INSTALL.md!
```
Comment (bhtucker) [inline bin/bootstrap.sh]: Comparing to INSTALL.md, this script (of course) lacks `requirements-dev` — should INSTALL.md have a note about that file being ignored in remote deployments, or is it obvious by convention?

File: bin/bootstrap.sh (hunk 1 above)
Comment (tvogels01) [reply bin/bootstrap.sh]: The `PyYAML` package needs the Python headers to compile.

File: bin/bootstrap.sh (hunk 2 above)
Comment (tvogels01) [reply bin/bootstrap.sh]: Thanks.  Added notes around that.

**Pull request merged**